### PR TITLE
[PWA-2400] Update packages and dependencies

### PR DESCRIPTION
**Pull request by jeremycharron**
## Description

Update packages and dependencies to reduce number of High+ vulnerabilities in yarn and npm audit
`yarn audit` vulnerabilities were reduced from 82 to 21
`npm audit` vulnerabilities were reduced from 78 to 17
Vulnerabilities related to Webpack 5, ESLint 8 and Storybook were out of scope as they have dedicated stories.

## Related Issue

Closes PWA-2400

### Verification Stakeholders

Project maintainers

### Specification

Dependencies from root package, create-pwa, pwa-buildpack, venia-concept, venia-integration-tests are impacted

**Still generating vulnerabilities - refactor required**
`@storybook/react`: 6.3.7 / 6.4.8 (To be updated in PWA-1867)
`webpack` : 4.46.0 / 5.65.0 (To be updated to 5 in PWA-1893)
`copy-webpack-plugin`: Updated to 6.4.0 / 10.0 (7.0 and adove requires Webpack5, tbd in PWA-1893).
`boxen`: 5.1.2 / 6.2.1 (Version 6 introduces breaking changes - now pure ESM imports)
`@pmmmwh/react-refresh-webpack-plugin`: 0.4.3 / 0.5.3

**Still generating vulnerabilities - module updated to latest version**
`danger`: 10.7.1 / 10.7.1 (published a month ago)
`graphql-cli`: 4.1.0 / 4.1.0 (published a year ago)
`@graphql-cli/validate`: 2.1.0 / 2.1.0 (published 2 years ago)
`eslint-plugin-package-json`: 0.1.4 / 0.1.4 (published 2 years ago)
`tap-diff`: 0.1.1 / 0.1.1 (published 6 years ago).

**No longer generating vulnerabilities**
`eslint`: Updated to 7.32.0 / latest 8.4.1 (Breaking changes - To be updated in PWA-2451)
`jest`: Updated to 26.6.3 / latest 27.4.3 (Version 27 has multiple breaking changes)
`chokidar`: Updated to 3.5.2
`coveralls`: Updated to 3.1.1
`graphql-config`: Updated to 3.4.1
`jest-junit`: Updated to 13.0.0
`bundlesize`: Updated to 0.18.1
`inquirer`: Updated to 8.2.0
`envalid`: Updated to 6.0.1
`yargs`: Updated to 17.3.0
`sharp`: Updated to 0.29.3

#### Test scenario(s) for direct fix/feature
- Run `yarn audit` and `npm audit` at root, and verify vulnerabilities have been reduced
- Perform general regression smoke test on project and updated packages
- Verify `yarn test` works properly
- Verify that `yarn run lint` and `yarn run prettier` still works
- Create scaffold project and verify `yarn build` and `yarn start` still works

#### Is Browser/Device testing needed?

No, device and browser testing is not needed

#### Any ad-hoc/edge case scenarios that need to be considered?

Not that I can think of, but I still have limited knowledge of the full solution, so any feedback is appreciated

## Screenshots / Screen Captures (if appropriate)
yarn audit develop
yarn audit 2400

## Breaking Changes (if any)

Should not have any breaking change

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js [hunk 1]
```diff
@@ -44,9 +44,6 @@ test('is a yargs builder', async () => {
     );
 
     expect(output).toMatch('Create a PWA');
-
-    // throws because it wants a positional argument--just checking
-    expect(() => createProjectCliBuilder.builder(yargs)).toThrow('positional');
```
Comment (jeremycharron) [inline packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: Actually not sure about this one. Tests are failing if we keep this. Any feedback is appreciated.

**Review by revanth0212 — CHANGES_REQUESTED** [review]
Everything seems fine, I have done a smoke test, scaffold test, lint test, prettier test, jest test and a cypress test.

Nice work @jeremycharron 

I do have couple of questions though, please check them out.

File: packages/venia-ui/lib/components/ProductOptions/__tests__/options.spec.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { createTestInstance } from '@magento/peregrine';
 import Options from '../options';
 
 jest.mock('../../../classify');
-jest.mock('uuid/v4', () => () => '00000000-0000-0000-0000-000000000000');
+jest.mock('uuid', () => () => '00000000-0000-0000-0000-000000000000');
```
[the remark this answers is not in the document]
Comment (jeremycharron) [reply packages/venia-ui/lib/components/ProductOptions/__tests__/options.spec.js:7]: Yes, unfortunately uuid/v4 is deprecated, we must use uuid now. Ref : https://www.npmjs.com/package/uuid#deep-requires-now-deprecated

File: packages/pwa-buildpack/lib/WebpackTools/plugins/UpwardIncludePlugin.js [hunk 3]
```diff
@@ -53,10 +53,13 @@ class UpwardIncludePlugin {
 
         debug('assets collection complete, %O', this.assetMap);
 
-        new CopyPlugin(Object.values(this.assetMap), {
-            copyUnmodified: true,
-            logLevel: 'error'
-        }).apply(this.compiler);
+        new CopyPlugin(
+            { patterns: Object.values(this.assetMap) },
```
[the remark this answers is not in the document]
Comment (jeremycharron) [reply packages/pwa-buildpack/lib/WebpackTools/plugins/UpwardIncludePlugin.js:57]: Yes, we need to have an object now. Ref : https://github.com/webpack-contrib/copy-webpack-plugin/releases/tag/v6.0.0

**Comment by jeremycharron** [comment]
Hello @revanth0212, thank you for your review. I answered your questions, but let me know if there is anything else!

File: packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js (hunk 1 above)
Comment (revanth0212) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: I thought it was an old obsolete test but unfortunately, it runs just fine in `develop`, so maybe something has changed in this PR:

Develop: 

image

**Review by revanth0212 — DISMISSED** [review]
The PR is almost there. Seems like the test you removed, actually passed in `develop`. Something to check. Ill try to read through the code to get some info.

File: packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js (hunk 1 above)
Comment (jeremycharron) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: Yes, the test only fails on this branch. If I recall, if was something about `.toThrow('positional');`

Comment (jeremycharron) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: Here's the error generated when running `yarn test` : 
image

Comment (jeremycharron) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: @pedrochiossi 
I'm seeing this as of [version 17.0.0](https://github.com/yargs/yargs/releases/tag/v17.0.0) : 
`.positional() now allowed at root level of yargs`

Comment (pedrochiossi) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: @jeremycharron I don't think this is related, I would look for something on builder function, but if seems it's not on their release notes.

Comment (jeremycharron) [reply packages/pwa-buildpack/lib/__tests__/cli-create-project.spec.js:49]: @tjwiebell has written a new test. I will get in touch with him so we can replace this one.

**Review by pedrochiossi — DISMISSED** [review]
All good @jeremycharron

**Comment by jeremycharron** [comment]
Scaffold build is now successful.

**Comment by mikhaelbois** [comment]
Here are the audit results on moment of QA.

**Project Root - Before**

`yarn audit`
88 vulnerabilities found - Packages audited: 1942
Severity: 60 Moderate | 28 High

`npm audit`
94 vulnerabilities (3 low, 49 moderate, 40 high, 2 critical)

**Scaffold - Before**

`yarn audit`
51 vulnerabilities found - Packages audited: 2543
Severity: 3 Low | 28 Moderate | 18 High | 2 Critical

`npm audit`
70 vulnerabilities (4 low, 36 moderate, 28 high, 2 critical)

---

**Project Root - After**

`yarn audit`
27 vulnerabilities found - Packages audited: 2015
Severity: 9 Moderate | 18 High

`npm audit`
75 vulnerabilities (3 low, 36 moderate, 34 high, 2 critical)


**Scaffold - After**

`yarn audit`
51 vulnerabilities found - Packages audited: 2543
Severity: 3 Low | 28 Moderate | 18 High | 2 Critical

`npm audit`
64 vulnerabilities (4 low, 31 moderate, 27 high, 2 critical)

**Comment by mikhaelbois** [comment]
@jeremycharron 
Following ran on Project Root ✅ 
```
yarn build
yarn lint
yarn prettier:check
yarn stage:venia
yarn storybook:venia
yarn test
yarn watch:venia
```

`yarn watch:all` works but the message visual in the console has a cosmetic issue.
Reporting it just in case.

Expected
Capture d’écran, le 2022-01-14 à 14 21 35

Current result
Capture d’écran, le 2022-01-14 à 14 27 56

Tested scaffolding with Yarn and NPM and everything worked fine for watch and build. ✅ 

Cypress tests ✅

**Comment by jeremycharron** [comment]
I pushed a fix for `yarn watch:all` visual being broken.
Boxen version 5 introduces text wrapping which doesn't seem to support long strings very well (such as URLs)
Proposed fix looks like this : 

image

**Review by supernova-at — APPROVED** [review]
Code looks good, running verification steps now

**Pull request merged**